x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -436,3 +437,105 @@ func TestCalcJoinPoolShares(t *testing.T) {
 		})
 	}
 }
+
+func TestRandomizedPoolInvariants(t *testing.T) {
+	type testCase struct {
+		initialTokensDenomIn  int64
+		initialTokensDenomOut int64
+
+		percentRatio int64
+
+		numShares sdk.Int
+	}
+
+	const denomOut = "denomOut"
+	const denomIn = "denomIn"
+
+	// generate test case with randomized initial assets and join/exit ratio
+	newCase := func() (tc *testCase) {
+		tc = new(testCase)
+		tc.initialTokensDenomIn = rand.Int63n(1_000_000)
```

Comment:
Should we instead use a source object, i.e. `rng := rand.NewSource(time.Now().Unix())`

Reply:
Replaced!